File: ROS2_WS/src/my_python_pkg/my_python_pkg/Publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class MyPublisher(Node):
    def __init__(self):
        super().__init__("publisher_node")

        self.name = "Iron Man"
        self.publisher_ = self.create_publisher(String, "publisher_node", 10)
        self.timer_ = self.create_timer(0.5, self.publish_message)
        self.get_logger().info("Publisher is publishing")


    def publish_message(self):
        msg = String()
        msg.data = "Hello this is " + str(self.name) + " from publiser node"
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = MyPublisher()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: ROS2_WS/src/my_cpp_package/src/publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class PublisherNode : public rclcpp::Node{
    public:
    PublisherNode() : Node("Publisher"), publisher_name_("My Publisher"){
        publisher_ = this->create_publisher<example_interfaces::msg::String>("publisher_node", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&PublisherNode::publishNews, this));
        RCLCPP_INFO(this->get_logger(), "Publisher Node has started!");
    }

    private:
    void publishNews(){
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hi this is ") + publisher_name_ + std::string(" from publishing node!");
        publisher_->publish(msg);
    }

    std::string publisher_name_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}